### PR TITLE
Provide a synchronous registration and unregistration of services.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Some parts of repl require being able to disable a service and block
until it's disabled.  Provide this API.

@metadave

### CODE REVIEW

File: src/riak_core_service_mgr.erl [hunk 1]
```diff
@@ -179,6 +195,15 @@ handle_call(get_stats, _From, State) ->
     PStats = [{Protocol, Count} || {Protocol,{_Stats,Count}} <- Stats],
     {reply, PStats, State};
 
+handle_call({register_service, Protocol, Strategy},  _From, State) ->
+    {{ProtocolId,_Revs},_Rest} = Protocol,
```
Comment (bookshelfdave) [inline src/riak_core_service_mgr.erl]: how would you feel about register_service + unregister_service calling the same shared functions?

Comment (cmeiklejohn) [inline src/riak_core_service_mgr.erl]: I feel pretty good about it.  Do you mind looking at the amended commit?

**Comment by bookshelfdave** [comment]
can register_service/unregister_service be removed in favor of all sync msgs?

**Comment by cmeiklejohn** [comment]
Not sure, but I would rather do that deprecation in 2.0 instead of 1.4.4.

**Comment by bookshelfdave** [comment]
+1

**Comment by cmeiklejohn** [comment]
I'm going to merge this and open a second PR to forward port this to 2.0, then I'll open another issue for tracking the deprecation of the async registration, sound good?

**Pull request merged**